helpers/embeds.js

```diff
@@ -0,0 +1,78 @@
+import { findComposedAncestor } from '../helpers/dom.js';
+import { html } from 'lit';
+import { tryGetPluginByKey } from '@brightspace-ui/core/helpers/plugins.js';
+
+class RenderNode {
+	#childNodes;
+	#node;
+
+	constructor(node) {
+		this.#node = node;
+		this.#childNodes = [];
+	}
+
+	addChildNode(childNode) {
+		this.#childNodes.push(childNode);
+	}
+
+	async renderNode(options) {
+		for await (const childNode of this.#childNodes) {
+			await childNode.renderNode(options);
+		}
```

Comment:
Ideally this shouldn't need to queue like this, since siblings should all be able to render "in parallel", but there's some timing wonkiness going on here with fetching the OAuth2 tokens that causes this promise not to resolve properly. It's a bit of a headache, so for now I'm just leaving it like this while I poke around.